signing/signing.go

```diff
@@ -1,20 +1,363 @@
 package signing
 
 import (
+	"bytes"
 	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
 	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http"
 	"path/filepath"
+	"reflect"
 	"strings"
+	"time"
 
+	"github.com/go-openapi/runtime"
+	"github.com/google/go-github/v42/github"
 	"github.com/gorilla/mux"
 	"github.com/ossf/scorecard/v2/cron/data"
+	"github.com/rhysd/actionlint"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	hashedrekord "github.com/sigstore/rekor/pkg/types/hashedrekord/v0.0.1"
+	rekord "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
 )
 
 type ScorecardOutput struct {
-	JsonOutput string
+	SarifOutput string
+	JsonOutput  string
+}
+
+// verifySignature receives the scorecard analysis payload, looks up its associated tlog entry and
+// certificate, and extracts the repository's workflow file to ensure its legitimacy.
+func VerifySignature(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "error reading http request body", http.StatusInternalServerError)
+		log.Println(err)
+		return
+	}
+
+	// Unmarshal request body.
+	var scorecardOutput ScorecardOutput
+	err = json.Unmarshal(reqBody, &scorecardOutput)
+	if err != nil {
+		http.Error(w, "error unmarshalling request body", http.StatusInternalServerError)
```

Comment:
can u follow the same pattern as other PRs (return err) and have one function that does the return?

Reply:
I already sort of did this (VerifySignature calls verifySignature), so currently VerifySignature only parses the request body and parameters, and then calls verifySignature to do the rest of the work.

Reply:
`verifySignature` looks great, +1
I think we can do the same for `VerifySignature()`? I'm thinking of having all functions just return `error` and the top one only do:
```
err, ... := someFunction()
if err != nil{
  http.Error(w, err.Error(), http.StatusInternalServerError)
  return
}

// All good
```

like we did for the other PRs (forgot which one it was)